frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+const { utils } = require( "ethers" );
+// import utils from "ethers";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{utils.formatEther(data.balance)}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
```

Comment:
I think, this could be a class which extends `React.Component`.

Reply:
See https://reactjs.org/docs/components-and-props.html. Either defining a component as as a function or as a class that extends React.Component is fine. I have seen the first approach used much more frequently.

Reply:
Also, as this is only used by the main component of the file, there is no need to put the small component in another file.

Reply:
ES6 Classes are just syntactic sugar representing constructor functions. Their only purpose is readability. Maybe, they are not so often used because web developers don't know them much or OOP. My personal advantage is: I can better separate code blocks from longish nested function definitions so that I need less time searching where the code block ends. Don't know, but it helps me to faster understand the code structure.

Also, functions with > 40 lines are often too long. Maybe `EthereumAccountDetails` is at the limit of what can be in a single function. But `Wallet` is already too long for a single function. A `class` would solve this issue.

Reply:
Better function name should be `EthereumAccountDetails`.

Reply:
Please read up on the difference of functional components and class components in React. There is a large difference in how they are written and which functionalities they provide. It is not about some syntactical sugar. Long term React developers know that they can define components as classes, but since the introduction of functional components and hooks rarely ever use them anymore. I do not support rewriting it as a class component. 

Also, as functional components define their own functions to handle different scenarios, I do not support restricting the length of a function to 40 lines as functional components require other functions to implement functionality. Restricting the return statement of a functional component however, or more specifically the number of JSX statements makes sense.